doc/sphinx/source/graph-basics.rst

```diff
@@ -1,24 +1,120 @@
+preferences when it comes to the above factors, is a very heavy burden 
+on framework owners.
+
+Adding the nGraph compiler to the system lightens that burden by raising the 
+abstraction level, and by letting any hardware-specific backends make these 
+decisions automatically. The nGraph Compiler is designed to be able to take into 
+account the needs of each target hardware platform, and to achieve maximum 
+performance.
+
+This makes things easier on framework owners, but also (as new models are developed) 
+on data scientists, who will not have to keep in mind nearly as many low-level 
+hardware details when architecting their models with layers of complexity for 
+anything other than a :abbr:`Just-in-Time (JIT)` compilation.     
+
+While the first generation frameworks tended to need to make a tradeoff between 
+being "specialized" and "adaptable" (the trade-off between training and inference), 
+nGraph Library permits algorithms implemented in a DNN to be both specialized 
+and adaptable. The new generation of software design in and around AI ecosystems 
+can and should be much more flexible.   
+
+
+* :ref:`framework_bridges`
+* :ref:`about_transformers`
+* :ref:`graph_shaping`
+ 
+
+
+.. _framework_bridges:
+
+Framework bridges
+=================
+
+In the nGraph ecosystem, a framework is what the data scientist uses to solve 
+a specific (and usually large-scale) deep learning computational problem with 
+the use of a high-level, data science-oriented language. 
+
+A framework :term:`bridge` is a software layer (typically a plugin *for* or an 
+extension *to* a framework) that translates the data science-oriented language 
+into a compute-oriented language called a :term:`data-flow graph`. The bridge 
+can then present the problem to the nGraph :abbr:`Abstraction Layer (AL)` which 
+is responsible for execution on an optimized backend by performing graph 
+transformations that replace subgraphs of the computation with more optimal 
+(in terms of machine code) subgraphs. Throughout this process, ``ops`` represent 
+tensor operations. 
+
+Either the framework can provide its own graph of functions to be compiled and 
+optimized via :abbr:`Ahead-of-Time (AoT)` compilation to send back to the 
+framework, or an entity (framework or user) who requires the flexibility of 
+shaping ops directly can use our graph construction functions to experiment with 
+building runtime APIs for their framework, thus exposing more flexible multi-theaded compute 
+power options to 
+
+See the section on :doc:`howto/execute` for a detailed walk-through describing 
+how this translation can be programmed to happen automatically via a framework. 
+
+
+.. _about_transformers:
+
+Transformer ops
+================
+
+A framework bridge may define its own bridge-specific ops, as long as they can be 
+converted to transformer ops. This is usually achieved by them first being 
+converted to core ops on the way. For example, if a framework has a 
```

Comment:
Have we defined core ops? If not, we can say that "core" ops are "a standard set of ops that all backends are required to support either directly or via translation"